### DISCUSSION TITLE
Compute area coverage during search

### DISCUSSION THREAD

**Discussion by g2giovanni** [stac-utils/stac-fastapi#633] [Q&A]
Hello everyone,

I am reaching out to inquire about the feasibility of automatically calculating the total geographical area covered by each item in search results, taking into account the input geometry specified in the query.

Specifically, I'm interested in understanding whether it's possible to determine the percentage of the input geometry that an item covers. For instance, if I search using a particular geographical input, I would like to know if an item, say item X, covers 50% of that specified input geometry.

Is there an existing feature or functionality that allows for this kind of calculation? Or would this require developing it from the ground up?

Any insights or guidance on this matter would be greatly appreciated.

Thank you very much.

**Comment by vincentsarago** [comment]
FYI: Sorry I converted this to `discussion` because it's not related to any issue with stac-fastapi 

That's an interesting use case, I wonder if this should be an extension to the STAC API specification. In term of realization, I think this should be much more efficient if done by each backend directly (e.g this kind of calculation will be faster in Postgres/PgSTAC). This could be a feature request on stac-fastapi-pgstac, but I still think that it should be following some kind of specification (because you'll update the feature properties returned 🤷 )
